### PR TITLE
ci: standardize dockerfiles

### PR DESCRIPTION
**Pull request by ernestojeda**
- [x] use alpine 3.12
- [x] common `apk add` where possible
- [x] update copyright header where possible

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/master/.github/CONTRIBUTING.md

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by soda480 — APPROVED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

File: Dockerfile.build [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #
 
 # This is the base build image for 
-ARG BASE=golang:1.15-alpine
+ARG BASE=golang:1.15-alpine3.12
```
Comment (cloudxxx8) [inline Dockerfile.build:18]: should we add --update to the `RUN apk add` command?

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Review by soda480 — APPROVED** [review]
LGTM

**Pull request merged**